src/patternfly/components/Form/styles.scss

```diff
@@ -0,0 +1,462 @@
+  --pf-c-form__helper-text--FontSize: var(--pf-global--FontSize--sm);
+
+  // Textarea
+  --pf-c-form__textarea--Padding: var(--pf-global--spacer--sm);
+
+  // Actions
+  --pf-c-form__form-actions--MarginRight: var(--pf-global--spacer--md);
+
+  // Tooltip
+  --pf-c-tooltip--MarginLeft: var(--pf-global--spacer--xs);
+
+  // Inline
+  --pf-c-form--MarginRight--inline: var(--pf-global--spacer--md);
+}
+
+// Base form
+.pf-c-form {
+
+  // Form group - pf-c-form__group
+  &__group {
+
+    // Inline layout
+    &.pf-m-inline {
+      display: flex;
+      flex-flow: row wrap;
+      > * {
+        flex: 0;
+        margin-right: var(--pf-c-form--MarginRight--inline);
+      }
+
+      // Label outside of pf-c-form__check
+      > .pf-c-form__label {
+        flex: 1 0 100%;
+        margin: 0;
+      }
+
+      .pf-c-form__label {
+        white-space: nowrap;
+      }
+    }
+  }
+
+  // Form label - pf-c-form__label
+  &__label {
+    display: flex;
+    align-items: baseline;
+    // using padding here to prevent flicker
+    padding-bottom: var(--pf-c-form__label--PaddingBottom);
+    margin: 0;
+    font-size: var(--pf-c-form__label--FontSize);
+
+    // Hide label selection on click
+    &::selection {
+      background: none;
+    }
+
+    // State - disabled
+    &.pf-m-disabled {
+      color: var(--pf-c-form__label--Color--disabled);
+    }
+
+    // Required
+    &.pf-m-required {
+      position: relative;
+
+      &::before {
+        position: absolute;
+        top: var(--pf-c-form__label--Top--required);
+        left: var(--pf-c-form__label--Left--required);
+        content: "\002A";
+      }
+    }
+  }
+
+  // Label group - pf-c-form__label-group
+  // Used when label contains elements like tooltips
+  &__label-group {
+    display: flex;
+
+    // Space 
+    > *:not(.pf-c-form__label) {
+      margin-left: var(--pf-c-form__label-group__item--MarginLeft);
+    }
+
+    &.pf-m-tooltip {
+      display: inline-flex;
+      align-items: first baseline;
+
+      > * {
+        margin-right: var(--pf-c-tooltip--MarginLeft);
+      }
+    }
+  }
+
+  // Text, textarea and select inputs
+  &__input[type="text"],
+  &__select select,
+  &__input[list],
+  &__textarea {
+    width: 100%;
+    padding: var(--pf-c-form__input--PaddingTop) var(--pf-c-form__input--PaddingRight) var(--pf-c-form__input--PaddingBottom) var(--pf-c-form__input--PaddingLeft);
+    font-size: var(--pf-c-form__input--FontSize);
+    line-height: var(--pf-c-form__input--LineHeight);
+    color: var(--pf-c-form__input--Color);
+    background: var(--pf-c-form__input--Background);
+    border: none;
+    box-shadow: inset 0 0 0 var(--pf-c-form__input--BorderSize) var(--pf-c-form__input--BorderColor);
+    transition: .2s;
+
+    // Hover
+    &:hover:not([disabled]) {
+      box-shadow: inset 0 0 0 var(--pf-c-form__input--BorderSize--hover) var(--pf-c-form__input--BorderColor);
+    }
+
+    // Focus
+    &:focus,
+    &:hover:focus {
```

Comment:
@jgiardino what do you think about this?

Reply:
Can we remove the `:not([disabled])` a few lines up, since the disabled styles are set a few lines down? If we do that, then we should only need `:focus`.